crates/state/src/chain_state.rs

```diff
@@ -49,22 +51,45 @@ pub struct Chainstate {
     pub(crate) finalized_epoch: EpochCommitment,
 
     /// Rollup's view of L1 state.
+    #[diff_override(l1::L1ViewStateDiff)]
     pub(crate) l1_state: l1::L1ViewState,
 
     /// Pending withdrawals that have been initiated but haven't been sent out.
+    #[diff_override(StateQueueDiff<bridge_ops::WithdrawalIntent>)]
     pub(crate) pending_withdraws: StateQueue<bridge_ops::WithdrawalIntent>,
 
     /// Execution environment state.  This is just for the single EE we support
     /// right now.
+    #[diff_override(exec_env::ExecEnvStateDiff)]
     pub(crate) exec_env_state: exec_env::ExecEnvState,
 
     /// Operator table we store registered operators for.
+    #[diff_override(StateQueueDiff<bridge_state::OperatorEntry>)]
     pub(crate) operator_table: bridge_state::OperatorTable,
 
     /// Deposits table tracking each deposit's state.
+    #[diff_override(StateQueueDiff<bridge_state::DepositEntry>)]
     pub(crate) deposits_table: bridge_state::DepositsTable,
```

Comment:
Using proc macros for this *would* be great, however we're planning on converting all of these chainstate defs over to be defined in terms of the Pythonic SSZ schemas, so this doesn't actually help us very much for mainnet.

Reply:
Wait, but why should a serialization scheme affect how we write/derive diffs? Is it going to replace the chainstate diff?

Reply:
We wouldn't be likely to have these structs that you're putting the proc macro derive / attributes onto, since they'd be code generated.

We could go and have the codegen also emit these attributes, but that interferes with separation of concerns and would get in the way if we need to introduce additional bookkeeping in (what's currently called) `StateCache`.